### DISCUSSION TITLE
Unrecognised mime type while saving image file in pocketbase

### DISCUSSION THREAD

**Discussion by rameshkumar70** [pocketbase/pocketbase#4849] [Q&A]
Hi, 
I'm trying to send a file, obtained from an html `input` of type "file" using a POST call using fetch/formdata. 
I am able to read the file, use canvas to resize the image, create a blob and then a new file out of it before shipping it out with a fetch/POST call.  Trying to update the collection in pocketbase gives me the this error, in the pocketbase dev trace:

`ERROR PATCH /api/collections/Student/records/v2yxx2q60izooey
└─ Something went wrong while processing your request.
   └─ profilePicture: "picture_qPqL4jJn5p.png" mime type must be one of: image/jpeg, image/png, image/svg+xml, image/gif, image/webp..`

Any idea what am I possibly doing wrong or not doing, anyone, please?

**Comment by ganigeorgiev** [comment]
Without a code sample I don't think anyone will be able to help you.

The above error indicates that your submitted file is none of the specified mimetypes and probably you are not constructing the blob correctly.

The mimetype detections works by reading the first couple bytes of the file and doesn't rely on its extension or the user submitted file type.

**Comment by rameshkumar70** [comment]
Here is the code, sorry:

I am using the function `processImage()` below to call the other two helper functions further down, `fileToDataUri()` and `resizeImage()`:

`const formData = new FormData(evt.currentTarget);`
`const imageBlob = await processImage(profilePictureInput.files[0]);`
`formData.set('profilePicture', new File([imageBlob],  "picture.png");`
`const response = await fetch("/students/id/personal", {`
`	method: 'POST',`
`	body: formData,`
`});`

`	async function processImage(file) {`
`		let img = new Image();`
`		img.src = await fileToDataUri(file);`
`		let resizedImageBlob;`
`		img.addEventListener('load', () => {`
`			resizedImageBlob = resizeImage(img);`
`		});`
`		return resizedImageBlob;`
`	}`

`	function fileToDataUri(field) {`
`		return new Promise((resolve) => {`
`			const reader = new FileReader();`
`			reader.addEventListener('load', () => {`
`				resolve(reader.result);`
`			});`
`			reader.readAsDataURL(field);`
`		});`
`	}`

`	async function resizeImage(img) {`
`		const width = 100;`
`		const height = 100;`
`		let canvas = document.createElement('canvas');`
`		let context = canvas.getContext('2d');`
`		canvas.width = width;`
`	        canvas.height = height;`
`		context.drawImage(img, 0, 0, width, height);`
`		const response = await fetch(canvas.toDataURL());`
`		const fileBlob = await response.blob();`
`		return fileBlob;`
`	}`

**Reply by ganigeorgiev** [reply]
The `processImage` function is wrong because it will return immediately. If you want the `img.addEventListener('load'` part inside it to have effect on the return statement you must resolve it as a `Promise` in a similar manner to `fileToDataUri` (_although I'm not sure what this does_).

In any case, this is not really related to PocketBase and while I want to help, general dev questions would be better addressed at StackOverflow or other similar forums.
